merlin/models/tf/loader.py

```diff
@@ -487,13 +506,44 @@ def _build_sparse_tensor(self, values, offsets, diff_offsets, num_rows, seq_limi
     def _handle_tensors(self, cats, conts, labels):
         to_return = super()._handle_tensors(cats, conts, labels)
 
-        for map_fn in self._map_fns:
-            to_return = map_fn(*to_return)
+        if len(self.label_names) > 1 and self.multi_label_as_dict:
+            X, y = to_return
+            to_return = X, dict(zip(self.label_names, y))
+
+        for _, transform in self._transforms:
+            to_return = transform(*to_return)
 
         return to_return
 
+    @property
+    def input_schema(self) -> Schema:
+        return self.data.schema
 
-class DatasetValidator(tf.keras.callbacks.Callback):
+    @property
+    def output_schema(self) -> Schema:
+        schema = self.input_schema
+
+        for to, transform in self._transforms:
+            if hasattr(transform, "compute_output_schema"):
+                _schema = transform.compute_output_schema(schema)
+            else:
+                raise ValueError(f"Couldn't infer schema from transform {transform}")
```

Comment:
If the `compute_output_schema()` of a Transform block is not implemented, could we not just assume it does not change the schema, instead of raising an exception?

Reply:
My impression so far is proper schema propagation through the layers/blocks is tricky, so I think being explicit is better here than silently assuming the default, but that's just my first impression. Do you think we should assume that the schema is not changed if the method is not implemented?